src/hotspot/share/runtime/arguments.cpp

```diff
@@ -892,74 +892,69 @@ static bool set_bool_flag(JVMFlag* flag, bool value, JVMFlagOrigin origin) {
   }
 }
 
-static bool set_fp_numeric_flag(JVMFlag* flag, char* value, JVMFlagOrigin origin) {
+static bool set_fp_numeric_flag(JVMFlag* flag, const char* value, JVMFlagOrigin origin) {
+  if (*value == '\0' || isspace(*value)) {
+    return false;
+  }
   char* end;
   errno = 0;
   double v = strtod(value, &end);
   if ((errno != 0) || (*end != 0)) {
     return false;
   }
+  if (g_isnan(v) || !g_isfinite(v)) {
```

Comment:
Surely the not-sign should not be there ???

Reply:
This is actually correct. If the number is NOT FINITE, we don't accept it. There's a test case for values like `"Infinity"`:

https://github.com/openjdk/jdk/blob/8413d87666e58c18c914b7df043ba3dbc6fa9022/test/hotspot/gtest/runtime/test_arguments.cpp#L280-L287

Reply:
Sorry my brain read that as `g_isinfinite()` - doh!